doc/rst/language/spec/data-parallelism.rst

```diff
@@ -752,44 +752,44 @@ side array expressions alias the left-hand side expression.
    are assigned to ``A`` may be read to compute the sum depending on the
    number of tasks used to implement the data parallel statement.
 
-.. _Indirect_Whole_Array_Indexing:
+.. _Promoted_Array_Indexing:
 
-Indirect Whole Array Indexing
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Promoted Array Indexing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Arrays can be indexed indirectly using an array of indices.
-Given an array ``A`` and an array of indices ``B`` in the domain of ``A``,
-the function call
+Array indexing operations can also be promoted.
+For example, an array of indices can be used to index into another array,
+as in the following expression:
 
 .. code-block:: chapel
 
-   f(A[B]);
+   A[B]
 
-is equivalent to
+which results in the promoted expression:
 
 .. code-block:: chapel
 
-   [b in B] f(A[b]);
+   [b in B] A[b]
 
-This is a legal expression only if the result of the promotion ``A[B]`` is not modified.
-The following statement
+However, it is an error to modify promoted expressions like this one.
+For example, the following is an error:
 
 .. code-block:: chapel
 
    A[B] += 3;
 
-is equivalent to
+If this was promoted, it would become the following:
 
 .. code-block:: chapel
 
    [b in B] A[b] += 3;
 
-which is not legal, as ``A`` cannot be modified without an explicit ``ref`` shadow variable.
-To modify the result of the promotion, the equivalent pattern can be used.
+This is illegal, as ``A`` cannot be modified without an explicit ``ref`` shadow
+variable. An explicit loop statement must be used, for example:
```

Comment:
Can you change "`ref` shadow variable" to "`ref` intent"?  I think of "shadow variables" as being the things a compiler introduces implicitly/invisibly that hide the user's original variable and don't tend to think of the term as applying when explicit `with` clauses are utilized (though I can see the potential relationship...).